File: projekty/PO_Projekt3/trawa.py
from roslina import Roslina

class Trawa(Roslina):
    def __init__(self, swiat, polozenie):
        super().__init__(swiat, polozenie, 0, 0, 0.10, 1)

    def tegoSamegoTypu(self, inny):
        return isinstance(inny, Trawa)

    def stworzDziecko(self, cel):
        return Trawa(self._swiat, cel)

    def rysowanie(self):
        return (170, 255, 0)

File: PO_Projekt2/ProgOb_Projekt2/src/main/java/pl/Dybowski/Adrian/Projekt2/WirtualnySwiat/Rosliny/Trawa.java
package pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Rosliny;

import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Organizm;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Stale;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Swiat;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Wspolrzedne2D;

public class Trawa extends Roslina {

	public Trawa(Swiat swiat, Wspolrzedne2D polozenie) {
		super(swiat, polozenie, Stale.sila.get(Trawa.class), Stale.inicjatywa.get(Trawa.class),
				Stale.szansaRozprzestrzeniania.get(Trawa.class), Stale.probyRozprzestrzeniania.get(Trawa.class));
	}

	@Override
	public boolean tegoSamegoTypu(Organizm inny) {
		return inny instanceof Trawa;
	}

	@Override
	public Organizm stworzDziecko(Wspolrzedne2D cel) {
		return new Trawa(swiat, cel);
	}

}
